pydantic/json_schema.py

```diff
@@ -1195,7 +1195,10 @@ def typed_dict_schema(self, schema: core_schema.TypedDictSchema) -> JsonSchemaVa
         with self._config_wrapper_stack.push(config):
             json_schema = self._named_required_fields_schema(named_required_fields)
 
-        extra = config.get('extra', 'ignore')
+        try:
+            extra = schema['extra_behavior']
+        except KeyError:
+            extra = config.get('extra', 'ignore')
```

Comment:
it would be faster to avoid the exception:

```suggestion
        extra = schema.get('extra_behavior')
        if extra is None:
            extra = config.get('extra', 'ignore')
```